src/com/sun/jna/Function.java

```diff
@@ -355,9 +369,11 @@ else if (args[i] instanceof PostCallRead) {
                 else if (Structure[].class.isAssignableFrom(inArg.getClass())) {
                     Structure.autoRead((Structure[])inArg);
                 }
+                /* Clear args */
+                args[i] = null;
```

Comment:
Is this because the `args` array is now being cached?  Might be worth a little more expansive note to that effect.